src/tax-lot/tax-lot.repo.ts

```diff
@@ -0,0 +1,175 @@
+} from "src/schema/tax-lot";
+import {
+  taxLot,
+  zoningDistrict,
+  zoningDistrictClass,
+  zoningDistrictZoningDistrictClass,
+} from "src/schema";
+import { MultiPolygon } from "geojson";
+
+export class TaxLotRepo {
+  constructor(
+    @Inject(DB)
+    private readonly db: DbType,
+  ) {}
+
+  #checkTaxLotByBbl = this.db.query.taxLot
+    .findFirst({
+      columns: {
+        bbl: true,
+      },
+      where: (taxLot, { eq, sql }) => eq(taxLot.bbl, sql.placeholder("bbl")),
+    })
+    .prepare("checkTaxLotByBbl");
+
+  async taxLotByBbl(bbl: string) {
+    let result: SelectTaxLotNested | undefined;
+    try {
+      result = await this.db.query.taxLot.findFirst({
+        columns: {
+          boroughId: false,
+          landUseId: false,
+          wgs84: false,
+          liFt: false,
+        },
+        where: (taxLot, { eq }) => eq(taxLot.bbl, bbl),
+        with: {
+          borough: true,
+          landUse: true,
+        },
+      });
+    } catch {
+      throw DataRetrievalException;
+    }
+
+    if (result === undefined) throw ResourceNotFoundException;
+    return result;
+  }
+
+  async taxLotByBblGeoJson(bbl: string) {
+    let result: SelectTaxLotSpatial | undefined;
+    try {
+      result = await this.db.query.taxLot.findFirst({
+        columns: {
+          bbl: true,
+          block: true,
+          lot: true,
+          address: true,
+        },
+        extras: {
+          geometry: sql<string>`ST_AsGeoJSON(${taxLot.wgs84}, 6)`.as(
+            "geometry",
+          ),
+        },
+        where: (taxLot, { eq }) => eq(taxLot.bbl, bbl),
+        with: {
+          borough: true,
+          landUse: true,
+        },
+      });
+    } catch {
+      throw DataRetrievalException;
+    }
+
+    if (result === undefined) throw ResourceNotFoundException;
+
+    const geometry = JSON.parse(result.geometry) as MultiPolygon;
+
+    return {
+      type: "Feature",
+      id: result.bbl,
+      properties: {
+        bbl: result.bbl,
+        borough: result.borough,
+        block: result.block,
+        lot: result.lot,
+        address: result.address,
+        landUse: result.landUse,
+      },
+      geometry,
+    };
+  }
+
+  async zoningDistrictByTaxLotBbl(bbl: string) {
+    let taxLotCheck: Pick<SelectTaxLot, "bbl"> | undefined;
+    try {
+      taxLotCheck = await this.#checkTaxLotByBbl.execute({ bbl });
+    } catch {
+      throw DataRetrievalException;
+    }
+    if (taxLotCheck === undefined) throw ResourceNotFoundException;
+    try {
+      const zoningDistricts = await this.db
+        .select({
+          id: zoningDistrict.id,
+          label: zoningDistrict.label,
+        })
+        .from(zoningDistrict)
+        .leftJoin(
+          taxLot,
+          sql`ST_Intersects(${taxLot.liFt}, ${zoningDistrict.liFt})`,
+        )
+        .where(eq(taxLot.bbl, bbl));
+
+      return {
+        zoningDistricts,
+      };
+    } catch {
+      throw DataRetrievalException;
+    }
+  }
```

Comment:
We'll actually want to split this into two different methods. The service will call the methods as it needs them. 

The first method will be a public "execution" of the private prepared statement

*Code segments are only for illustrative purposes. Please verify functionality in full coding environment*
```typescript
async checkTaxLotByBbl(bbl: string) {
 try {
      return this.#checkTaxLotByBbl.execute({ bbl });
    } catch {
      throw DataRetrievalException;
    }
}
```    
The second method can keep the same name. But, it will only be concerned about actual query for zoningDistricts

```typescript
async zoningDistrictByTaxLotBbl(bbl: string) {
   try {
      return await this.db
        .select({
          id: zoningDistrict.id,
          label: zoningDistrict.label,
        })
        .from(zoningDistrict)
        .leftJoin(
          taxLot,
          sql`ST_Intersects(${taxLot.liFt}, ${zoningDistrict.liFt})`,
        )
        .where(eq(taxLot.bbl, bbl));
    } catch {
      throw DataRetrievalException;
    }
  }
}
```

The service will then control the flow of calling the repositories
```typescript
async findZoningDistrictByTaxLotBbl(bbl: string) {
  const taxLotCheck = await this.taxLotRepo.checkTaxLotByBbl(bbl);
  if (taxLotCheck=== undefined) throw ResourceNotFoundException;
  return await this.taxLotRepo.zoningDistrictClassByTaxLotBbl(bbl)
}
```